linked_list/linked_list.py

```diff
@@ -0,0 +1,203 @@
+        self.size = 0
+
+    def add_front(self, data):
+        # Inserts element at first position.
+        n = node(data)
+        n.next = self.head
+        self.head = n
+        self.size += 1
+
+    def add_last(self, data):
+        # Inserts element at last position.
+        if not self.head:
+            n = node(data)
+            self.head = n
+            self.size = 1
+            return
+        else:
+            n = self.head
+            while n.next is not None:
+                n = n.next
+            new_node = node(data)
+            n.next = new_node
+            self.size += 1
+            return
+
+    def add(self, data, index):
+        # Inserts data at index.
+        if index < 0 or index > self.size:
+            raise IndexError
+        temp = self.head
+        if index is 0:
+            add_front(data)
+            return
+
+        else:
+            for _ in range(index - 1):
+                temp = temp.next
+            n = node(data)
+            n.next = temp.next
+            temp.next = n
+            self.size += 1
+            return
+
+    def remove_front(self):
+        # Deletes First Element
+        n = self.head
+        if self.is_empty():
+            raise ValueError
+        removed = n.data
+        n = n.next
+        self.size -= 1
+        return removed
+
+    def remove_last(self):
+        # Deletes Last Element
+        n = self.head
+        if self.is_empty():
+            raise ValueError
+        if n.next is None:
+            removed = n.data
+            n = None
+        else:
+            temp = n
+            while temp.next.next is not None:
+                temp = temp.next
+            removed = temp.next.data
+            temp.next = None
+        self.size -= 1
+        return removed
+
+    def remove(self, index):
+        # Deletes Element present at index
+        if self.is_empty():
+            raise ValueError
+        if index < 0 or index >= self.size:
+            raise IndexError
+        if index > 0:
+            temp = self.head
+            for _ in range(index - 1):
+                temp = temp.next
+            removed = temp.next.data
+            temp.next = temp.next.next
+            self.size -= 1
+            return removed
+
+    def search(self, inp):
+        # Search for inp in Linked List and return it's index
+        temp = self.head
+        for i in range(self.size):
+            if temp.data == inp:
+                return i
+            else:
+                temp = temp.next
+        raise ValueError
+
+    def is_empty(self):
+        # Checks if Linked List is Empty.
+        return self.size is 0
+
+    def print_list(self):
+        # Prints Linked List
+        n = self.head
+        print('Linked List : [', end="")
+        while n:
+            if n.next:
+                print(n, end=", ")
+            else:
+                print(n, end="")
+            n = n.next
+        print(']')
+
+    def print_size(self):
```

Comment:
Implement `__len__` instead. 
Resource https://stackoverflow.com/questions/15114023/using-len-and-def-len-self-to-build-a-class